### ISSUE TITLE
Cannot Tab into TreeView After Filtering Removes Focused Node

### ISSUE THREAD

**Issue by yanmariomenev** [telerik/kendo-angular#4658]
### Description

Tab navigation skips the TreeView after filtering when the previously focused node is removed. This affects keyboard accessibility and appears to be caused by the roving tabindex referencing a non-existent element.

### Steps To Reproduce

1. Open any filter demo from our [article](https://www.telerik.com/kendo-angular-ui/components/treeview/filtering#filtering-using-the-hierarchy-binding-directive).
2. Use the keyboard to focus the "Decor" node in the TreeView.
3. Press "Shift + Tab" to move focus to the filter input.
4. Type "Armchair" in the filter input.
5. Press "Tab" to move forward in the tab order.

### Screenshots or video

[https://github.com/user-attachments/assets/f1606884-cb84-42f8-9f70-1756763118a0](https://github.com/user-attachments/assets/f1606884-cb84-42f8-9f70-1756763118a0)

### Actual Behavior

TreeView is skipped in the tab order. Focus moves past it.

### Expected Behavior

Focus should move into the filtered TreeView. Something like the recording or first node:

[https://github.com/user-attachments/assets/c5054918-5894-42aa-bc82-cdad6092cc93](https://github.com/user-attachments/assets/c5054918-5894-42aa-bc82-cdad6092cc93)

### Workaround

Handle **filterStateChange** event and update the tab index of the tree nodes:

[https://stackblitz.com/edit/angular-nwtttn83-c5gq1z6c?file=src%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/angular-nwtttn83-c5gq1z6c?file=src%2Fapp%2Fapp.component.ts)

### Browser

Chrome

### Browser version

latest

### OS type

MacOS

### Last working version of the Kendo UI for Angular package (if regression).

*No response*
